setup_builder.py

```diff
@@ -249,7 +260,7 @@ def _get_version(self) -> str:
         race condition, see:
         https://stackoverflow.com/a/2073599/13156561
         """
-        with open(self.pkg_path + "/__init__.py") as f:
+        with open(self.pkg_path + "/__init__.py", "r") as f:
```

Comment:
This one I'm ambivalent about.

I really like the simplicity and clarity of `self.pkg_path + "/__init__.py"`, but it jumps out at me as something that should go through `os.path.join`. Doing that seems more correct, but it makes the code less comprehensible. Given the nature of how this is used, I'd consider leaving as-is (choose clarity over purity).

If you do go the `os.path.join` route, be sure to make it `__init__.py` (get rid of the leading `/`) because Python's join allows absolute paths to erase previous path components.

Reply:
I agree. Following best practices is best